src/atomate2/common/flows/mpmorph.py

```diff
@@ -0,0 +1,518 @@
+    md_maker: Maker | None = None
+    postprocessor: EOSPostProcessor = field(default_factory=MPMorphPVPostProcess)
+    initial_strain: float | tuple[float, float] = 0.2
+    min_strain: float = 0.5
+    max_attempts: int | None = 20
+    energy_average_frames: int = 1
+
+    @job
+    def make(
+        self,
+        structure: Structure,
+        prev_dir: str | Path | None = None,
+        working_outputs: dict[str, Any] | None = None,
+    ) -> Flow:
+        """
+        Run an NVT+EOS equilibration flow.
+
+        Parameters
+        ----------
+        structure : Structure
+            structure to equilibrate
+        prev_dir : str | Path | None (default)
+            path to copy files from
+        working_outputs : dict or None
+            contains the outputs of the flow as it recursively updates
+
+        Returns
+        -------
+        .Flow, an MPMorph flow
+        """
+        if working_outputs is None:
+            if isinstance(self.initial_strain, float | int):
+                self.initial_strain = (
+                    -abs(self.initial_strain),
+                    abs(self.initial_strain),
+                )
+            linear_strain = np.linspace(
+                *self.initial_strain, self.postprocessor.min_data_points
+            )
+            working_outputs = {
+                "relax": {
+                    key: [] for key in ("energies", "volume", "stress", "pressure")
+                }
+            }
+
+        else:
+            working_outputs["relax"]["energy"] = [
+                sum(frame[-self.energy_average_frames :]) / self.energy_average_frames
+                for frame in working_outputs["relax"]["energies"]
+            ]
+
+            self.postprocessor.fit(working_outputs)
+            working_outputs = dict(self.postprocessor.results)
+            for k in ("pressure", "energy"):
+                working_outputs["relax"].pop(k, None)
+
+            if (
+                working_outputs.get("V0") is None
+            ):  # breaks whole flow here if EOS is not fitted properly
+                return Response(output=working_outputs, stop_children=True)
+            if (
+                working_outputs.get("V0") <= working_outputs.get("Vmax")
+                and working_outputs.get("V0") >= working_outputs.get("Vmin")
+            ) or (
+                self.max_attempts
+                and (
+                    len(working_outputs["relax"]["volume"])
+                    - self.postprocessor.min_data_points
+                )
+                >= self.max_attempts
+            ):
+                # If the equilibrium volume is within the range of fitted volumes,
+                # or if the maximum number of attempts has been performed, stop
+                # and return structure at estimated equilibrium volume
+                final_structure = structure.copy()
+                final_structure.scale_lattice(working_outputs["V0"])
+                return final_structure
```

Comment:
Consider

```python
            # Fit EOS to the running volume and energy from previous calculations
            self.postprocessor.fit(working_outputs)
            working_outputs = dict(self.postprocessor.results)

            # breaks whole flow here if EOS is not fitted properly
            if working_outputs.get("V0") is None:
                return Response(output=working_outputs, stop_children=True)

            # Check if the number of calculations has reached the maximum
            n_calcs = len(working_outputs["relax"]["volume"])
            max_calcs = self.postprocessor.min_data_points + (self.max_attempts or 1000)
            max_calcs_reached = n_calcs >= max_calcs

            v_0 = working_outputs.get("V0")
            v_min = working_outputs.get("Vmin")
            v_max = working_outputs.get("Vmax")

            # return final structure if calculated volume is in interpolation regime
            if max_calcs_reached or (v_min <= v_0 <= v_max):
                final_structure = structure.copy()
                final_structure.scale_lattice(v_0)
                return Response(output=final_structure)

            # Else, if the extrapolated equilibrium volume is outside the range of
            # fitted volumes, scale appropriately
            v_ref = v_min if v_0 < v_min else v_max
            eps_0 = (v_0 / v_ref) ** (1 / 3) - 1.0

            linear_strains = [np.sign(eps_0) * (abs(eps_0) + self.min_strain)]
```

Reply:
LGTM, I might save intermediate variables for the `if` statements but ultimately a matter of taste.